packages/hawtio/src/plugins/shared/attributes/AttributeTable.tsx

```diff
@@ -0,0 +1,76 @@
+import { useContext, useEffect, useState } from 'react'
+import { Card, CardBody, Text } from '@patternfly/react-core'
+import { Table, TableBody, TableHeader, TableProps } from '@patternfly/react-table'
+import { PluginNodeSelectionContext } from '@hawtiosrc/plugins/selectionNodeContext'
+import { AttributeValues } from '@hawtiosrc/plugins/connect/jolokia-service'
+import { attributeService } from './attribute-service'
+import './AttributeTable.css'
+import { NodeNameTable } from './NodeNameTable'
+
+export const AttributeTable: React.FunctionComponent = () => {
+  const { selectedNode } = useContext(PluginNodeSelectionContext)
+  const [attributes, setAttributes] = useState<AttributeValues[]>([{}])
+  const [isReading, setIsReading] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (!selectedNode) {
+      return
+    }
+
+    const childrenMbeansAttributes: AttributeValues[] = []
+    const readAttributes = async () => {
+      setIsReading(true)
+      if (selectedNode.children)
+        for (const mbean of selectedNode.children) {
+          const objectName = mbean.objectName
+          if (objectName) {
+            const attrs = await attributeService.read(objectName)
+            childrenMbeansAttributes.push(attrs)
+          }
+        }
+      setAttributes([...childrenMbeansAttributes])
+      setIsReading(false)
+    }
+    readAttributes()
+  }, [selectedNode])
+
+  if (!selectedNode) {
+    return null
+  }
+
+  if (isReading) {
+    return (
+      <Card>
+        <CardBody>
+          <Text component='p'>Reading attributes...</Text>
+        </CardBody>
+      </Card>
+    )
+  }
+
+  const tidyLabels = (str: string) =>
```

Comment:
This looks ad hoc. In the legacy Hawtio, we created a dedicated util function to share the common logic. Also write tests for the function.
https://github.com/hawtio/hawtio-core/blob/main/src/app/utilities/baseHelpers.ts#L479-L494

Reply:
A bit, I saw the old util function but because it was only used here it felt best to leave it here. I will move it to an util function and write tests.

Reply:
Yes, I did, and had to reimplement it because it failed in a few corner cases from my testing

We would need another one for snake-case too, but that should be more simple